contrib/auto-render/splitAtDelimiters.js

```diff
@@ -27,76 +27,72 @@ const findEndOfMath = function(delimiter, text, startIndex) {
     return -1;
 };
 
-const splitAtDelimiters = function(startData, leftDelim, rightDelim, display) {
-    const finalData = [];
+const escapeRegex = function(string) {
+    return string.replace(/[-/\\^$*+?.()|[\]{}]/g, "\\$&");
+};
+
 
-    for (let i = 0; i < startData.length; i++) {
-        if (startData[i].type === "text") {
-            const text = startData[i].data;
+const splitAtDelimiters = function(text, delimiters) {
+    let regexLeft;
+    for (let i = 0; i < delimiters.length; i++) {
+        if (i === 0) {
+            regexLeft = "(";
+        } else {
+            regexLeft += "|";
+        }
+        regexLeft += escapeRegex(delimiters[i].left);
+    }
+    regexLeft = new RegExp(regexLeft + ")");
 
-            let lookingForLeft = true;
-            let currIndex = 0;
-            let nextIndex;
+    let lookingForLeft = true;
+    let index;
+    const data = [];
 
-            nextIndex = text.indexOf(leftDelim);
-            if (nextIndex !== -1) {
-                currIndex = nextIndex;
-                finalData.push({
+    while (true) {
+        if (lookingForLeft) {
+            index = text.search(regexLeft);
```

Comment:
Using `match` could save a bit of work later since it returns both the index of the delimiter was well as which delimiter was matched although you'd still have to search through `delimiters` to find the matching right delimiter.

Reply:
I disagree. `match` is slower and produces lots of useless information. and as you pointed out a search is still needed.